src/bokeh/models/plots.py

```diff
@@ -312,10 +312,14 @@ def add_tools(self, *tools: Tool) -> None:
 
         '''
         for tool in tools:
-            if not isinstance(tool, Tool):
-                raise ValueError("All arguments to add_tool must be Tool subclasses.")
-
-            self.toolbar.tools.append(tool)
+            if isinstance(tool, str):
+                tool_obj = Tool.from_string(tool)
+            elif isinstance(tool, Tool):
+                tool_obj = tool
+            else:
+                raise ValueError(f"expected a string or Tool instance, got {tool!r}")
+
+            self.toolbar.tools.append(tool_obj)
```

Comment:
All of this and related tests seem unrelated, and therefore better in a dedicated PR with an issue to document an expanded API for the changelog